### PR TITLE
feat(webmail): Give folders and messages URL fragments

### PR DESCRIPTION
**Pull request by tadzik**
This allows linking to specific folders/messages
as well as going back and forth between messages with browser buttons.

It also attempts to fix our e2e tests to make them pass (or fail) more consistently.

### CODE REVIEW

**Comment by tadzik** [comment]
https://github.com/runbox/runbox7/issues/238 should be a consideration here: right now all the links are created on-the-fly on the JS side, making it impossible to open things in new tabs.

**Comment by gtandersen** [comment]
Nice @tadzik!

**Comment by petersalomonsen** [comment]
the cypress tests look very efficient, and I like the introduction of URL fragments for folders and messages.

File: src/app/app.component.ts [hunk 1]
```diff
@@ -151,24 +152,27 @@ export class AppComponent implements OnInit, AfterViewInit, CanvasTableSelectLis
...
-              }
-              if (combo === 'j') {
-                  this.canvastable.scrollDown();
-              }
-              const e: ExtendedKeyboardEvent = event;
-              e.returnValue = false;
-              return e;
-          }));
+    private _hotkeysService: HotkeysService
+  ) {
```
Comment (castaway) [inline src/app/app.component.ts:156]: Is this change of indentation on purpose?

Comment (tadzik) [reply src/app/app.component.ts:156]: It is – I switched it to 2-spaces to be like the rest of the file, since it was off before that.

Comment (castaway) [reply src/app/app.component.ts:156]: Its still 4 spaces...

Comment (tadzik) [reply src/app/app.component.ts:156]: Well, it's 4 spaces from column 0, but 2 from the outer scope... I think? :)

The empty line is silly indeed, I can remove it.

File: src/app/app.component.ts [hunk 2]
```diff
@@ -577,19 +598,20 @@ export class AppComponent implements OnInit, AfterViewInit, CanvasTableSelectLis
       this.selectedRowIds[this.selectedRowId] = true;
 
       if (!multiSelect && columnIndex > 1) {
+        let messageId: number;
         if (this.showingSearchResults) {
-          this.singlemailviewer.expectedMessageSize = this.searchService.api.getNumericValue(this.selectedRowId, 3);
```
Comment (castaway) [inline src/app/app.component.ts:581]: Confused here where expectedMessageSize went ?

**Comment by tadzik** [comment]
Replaced by https://github.com/runbox/runbox7/pull/612

File: src/app/app.component.ts (hunk 2 above)
Comment (tadzik) [reply src/app/app.component.ts:581]: Hopefully described a bit better in https://github.com/runbox/runbox7/pull/612/commits/75616816347eb35ad40e5dc71479ba1053036c72 :)

File: src/app/app.component.ts (hunk 1 above)
Comment (tadzik) [reply src/app/app.component.ts:156]: Commited more explicitely in https://github.com/runbox/runbox7/pull/612/commits/ad923afeb1f09495749d1ad37b164044c3b04900

**Pull request closed without merging**